src/ufuncs.c

```diff
@@ -192,6 +192,16 @@ static void *GetExteriorRing(void *context, void *geom) {
     return ret;
 }
 static void *get_exterior_ring_data[1] = {GetExteriorRing};
+/* the normalize funcion acts inplace */
+static void *GEOSNormalize_r_with_clone(void *context, void *geom) {
+    int ret;
+    void *new_geom = GEOSGeom_clone_r(context, geom);
+    if (new_geom == NULL) { return NULL; }
+    ret = GEOSNormalize_r(context, new_geom);
+    if (ret == -1) { return NULL; }
```

Comment:
You should destroy the ‘new_geom’ in this case

Reply:
Ah, yes, thanks for the catch, updated.